src/python/build_management/revisions.py

```diff
@@ -700,3 +701,24 @@ def write_revision_to_revision_file(revision_file, revision):
 def revision_pattern_from_build_bucket_path(bucket_path):
   """Get the revision pattern from a build bucket path."""
   return '.*' + os.path.basename(bucket_path)
+
+
+@memoize.wrap(memoize.FifoOnDisk(DISK_CACHE_SIZE))
+@memoize.wrap(memoize.Memcache(60 * 60 * 24 * 30))  # 30 day TTL
+def revision_to_branched_from(uri, revision):
+  """Interrogates git code review server to find the branch-from
+  revision of a component."""
+  full_uri = "%s/+/%s?format=JSON" % (uri, revision)
+  url_content = _get_url_content(full_uri)
+  # Hatefully, gerrit returns nonsense in the first line.
+  url_content = '\n'.join(url_content.splitlines()[1:])
+  result = _to_json_dict(url_content)
+  if not result:
+    return None
```

Comment:
add a logs.log_error() here and in all the subsequent error conditions so we don't silently break if things ever change.